bson/__init__.py

```diff
@@ -377,6 +377,31 @@ def _get_decimal128(data, position, dummy0, dummy1, dummy2):
     BSONMAX: lambda v, w, x, y, z: (MaxKey(), w)}
 
 
+_BSON_TYPE_MARKER_TO_PYTHON_TYPE_MAP = {
```

Comment:
I don't think you need this. You can just do type(value) on the value returned from the standard decoder. I'm guessing you're doing this to avoid the type() call. Is that more expensive than dict lookup?

Reply:
Just a lapse in judgment; I'll rewrite this to use `type(value)`.